lib/compat/wordpress-6.9/post-data-block-bindings.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Post Data source for the block bindings.
+ *
+ * @since 6.9.0
+ * @package gutenberg
+ * @subpackage Block Bindings
+ */
+
+/**
+ * Gets value for Post Data source.
+ *
+ * @since 6.9.0
+ * @access private
+ *
+ * @param array    $source_args    Array containing source arguments used to look up the override value.
+ *                                 Example: array( "key" => "foo" ).
+ * @param WP_Block $block_instance The block instance.
+ * @return mixed The value computed for the source.
+ */
+function gutenberg_block_bindings_post_data_get_value( array $source_args, $block_instance ) {
+	if ( empty( $source_args['key'] ) ) {
+		return null;
+	}
+
+	if ( empty( $block_instance->context['postId'] ) ) {
+		return null;
+	}
+	$post_id = $block_instance->context['postId'];
+
+	// If a post isn't public, we need to prevent unauthorized users from accessing the post data.
+	$post = get_post( $post_id );
+	if ( ( ! is_post_publicly_viewable( $post ) && ! current_user_can( 'read_post', $post_id ) ) || post_password_required( $post ) ) {
+		return null;
+	}
+
+	if ( 'date' === $source_args['key'] ) {
+		return esc_attr( get_the_date( 'c', $post_id ) );
+	}
+
+	if ( 'modified' === $source_args['key'] ) {
+		/*
+		 * Only return the modified date if it is later than the publishing date.
+		 */
```

Comment:
```suggestion
		// Only return the modified date if it is later than the publishing date.
```

Per single line comment https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-1-single-line-comments